subsys/bluetooth/host/gatt.c

```diff
@@ -274,6 +277,62 @@ static void sc_process(struct k_work *work)
 	atomic_set_bit(sc->flags, SC_INDICATE_PENDING);
 }
 
+#if defined(CONFIG_BT_SETTINGS_CCC_STORE_ON_WRITE)
+static struct gatt_ccc_store {
+	struct bt_conn *conn_list[CONFIG_BT_MAX_CONN];
+	struct k_delayed_work work;
+} gatt_ccc_store;
+
+static bool gatt_ccc_is_conn_queued(const struct bt_conn *conn)
+{
+	for (size_t i = 0; i < CONFIG_BT_MAX_CONN; i++) {
```

Comment:
I think you can eliminate all the for-loops in this PR by using `bt_conn_id_get()` to have each connection object use a pre-defined array index.

Reply:
Do you mean `bt_id_get()` ? So I would get an array of addresses, indexed with `id`, and I could use the address from this array instead of `conn->le.dst`. Does `bt_id_get()` always return all currently connected addresses?

Can we do this improvement in a follow-up PR and merge it now? This is a working solution. This PR fixes a bug that breaks many of our test cases and prevents us from discovering other problems. So it would be good to have it already and change the implementation in subsequent PR.

Reply:
All bt_conn objects are stored in an array inside conn.c. bt_id_get() returns the index to that array. You have an array of equal size of bt_conn pointers, so just use the return value of bt_id_get() to decide where to store each conn, instead of doing the for loop. This is a super simple change, so I'd rather do it now rather than as a follow-up. Basically you'll end up with stuff like this:
```
gatt_ccc_store.conn_list[bt_id_get(conn)] = bt_conn_ref(conn);
```